interface/repository/pokemon_csv_repository.go

```diff
@@ -0,0 +1,74 @@
+package repository
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/LuisTejedaS/ondemand-go-bootcamp/domain/model"
+	"github.com/LuisTejedaS/ondemand-go-bootcamp/infraestructure/data"
+)
+
+type pokemonCSVRepository struct {
+	csvDataSource data.CsvDataSource
+	pokemons      []*model.Pokemon
+}
+
+func NewpokemonCSVRepository(ds data.CsvDataSource) (*pokemonCSVRepository, error) {
+	adapter := &pokemonCSVRepository{csvDataSource: ds}
+
+	if err := adapter.LoadPokemons(); err != nil {
+		return nil, err
+	}
+
+	return adapter, nil
+}
+
+func (a *pokemonCSVRepository) LoadPokemons() error {
+	csvRecords, err := a.csvDataSource.ReadCollection()
+	a.pokemons = make([]*model.Pokemon, 0)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range csvRecords {
+		p := model.Pokemon{}
+
+		id, err := strconv.ParseUint(v[0], 10, 32)
+		if err != nil {
+			return err
+		}
+
+		p.ID = uint(id)
+		p.Name = v[1]
+
+		a.pokemons = append(a.pokemons, &p)
+	}
+
+	return nil
+}
+
+// FindById searches for a pokemon with the given id parameter.
+//
+// If the search is successful, a pointer to the found Pokemon is returned.
+// Otherwise and ErrPokemonNotFoundByID error is returned.
+func (a *pokemonCSVRepository) FindById(p *model.Pokemon, id int) (*model.Pokemon, error) {
```

Comment:
`FindById` and `FindAll` receives some pointer parameters (`p *model.Pokemon` and `p []*model.Pokemon`) that are not being used. Please remove  the function parameters. 

Also, please correct me if I'm wrong, but there is no real need to return pointers, returning the values should work just fine